src/applications/discharge-wizard/utilities/page-navigation.js

```diff
@@ -93,31 +93,53 @@ export const navigateForward = (
           !formResponses[nextShortName] &&
           editMode
         ) {
-          updateRouteMap([...routeMap, ROUTES?.[nextShortName]]);
+          if (routeMap[routeMap.length - 1] !== ROUTES?.[nextShortName]) {
```

Comment:
I think we could benefit from a comment(s) explaining what's going on here briefly, and/or smaller utility functions that `navigateForward` uses to help keep things more readable.